File: 7-kyu/the-skiponacci-sequence/javascript/solution.js
function* genFibonacci() {
  yield 1;
  let a = 0;
  let b = 1;
  let c = a + b;
  while (true) {
    yield c;
    a = b;
    b = c;
    c = a + b;
  }
}

function skiponacci(n) {
  let i = 0;
  const sequence = [];
  const fib = genFibonacci();
  while (i < n) {
    const nextFib = fib.next().value;
    const toPush = i % 2 === 1 ? "skip" : nextFib;
    sequence.push(toPush);
    i++;
  }
  return sequence.join(" ");
}

File: 7-kyu/the-skiponacci-sequence/python/solution.py
def skiponacci(n):
    a = 0
    b = 1
    skips = ['1']
    i = 1
    while i < n:
        nxt = f'{a + b}' if i % 2 == 0 else 'skip'
        a, b = b, a + b
        skips.append(nxt)
        i += 1
    return ' '.join(skips)